### PR TITLE
Fix incognito icon in cookie settings

### PR DESCRIPTION
**Pull request by mherrmann**
Fixes: https://github.com/brave/brave-browser/issues/11738

@mihaiplesa can you suggest additional reviewers?

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by mherrmann** [comment]
The [Linux build failure](https://ci.brave.com/job/pr-brave-browser-mherrmann-fix-incognito-icon-in-settings-linux/1/execution/node/249/log/) seems to be unrelated to this PR. I tried the generated .deb and am getting the expected UI change:

image

**Comment by mihaiplesa** [comment]
That icon looks familiar, wonder if we already had it added and we can just reference it.

Screen Shot 2020-11-06 at 11 14 18 AM

**Comment by mherrmann** [comment]
I did a preliminary search for the existing icon while researching the fix. Will try again.

**Comment by mherrmann** [comment]
It's available as incognito.icon and incognito_profile.icon in [vector_icons/chrome/app/vector_icons](https://github.com/brave/brave-core/tree/master/vector_icons/chrome/app/vector_icons). 

In Brave: 
image

.icon file previews with https://github.com/sadrulhc/vector-icons:

image

image

I am not sure whether we can use these .icon files in brave://settings. Do you know who we could ask @mihaiplesa?

File: patches/chrome-browser-resources-settings-site_settings-site_list_entry.html.patch
Comment (mkarolin) [inline patches/chrome-browser-resources-settings-site_settings-site_list_entry.html.patch:9]: @petemill can this be done with //brave/browser/resources/settings/brave_overrides/icons.js override instead of patching?

**Review by petemill — CHANGES_REQUESTED** [review]
Thanks for this, left some feedback that would result in not needing to patch.

**Comment by mherrmann** [comment]
Thank you very much for your kind review and suggestions @petemill @mkarolin. I implemented your recommended changes and still get the same screenshots:

image

image

I also verified that it uses the shorter SVG suggested by @petemill.

image

This is my first PR to brave-core. How should I proceed? For example, should I squash my commits, or rebase them onto the current `master`?

**Review by petemill — APPROVED** [review]
Looks perfect. Yes, please squash before merge if you are able.

You don't have to rebase on master as there are no conflicts and CI has passed enough steps apart from what seems like a transient error that IMO we can ignore.

Good to merge if your squash results in no code change.

**Comment by mherrmann** [comment]
Squashed and merged. Thank you @petemill @mkarolin @mihaiplesa !

**Comment by bsclifton** [comment]
🕶️

**Pull request merged**
